docs/vm-configuration.md

```diff
@@ -0,0 +1,265 @@
+kubevirt without requiring any user specification. For example, there is no need
+for the user to specify what VNC/SPICE port to use, or where to find the UEFI
+BIOS files. It is sufficient to know that VNC or UEFI must be enabled. Other
+areas of backend configuration will require the user to specify resources in an
+abstracted manner. For example, instead of providing a full file path for a disk,
+the tenant would provide the name of a PersistentVolumeClaim resource, or
+alternative k8s API resource. Further areas of backend configuration will
+require users to specify desired goals, which kubevirt then translates into
+specific configuration information. For example, the tenant may request exclusive
+use of CPUs, and kubevirt has to decide which host CPUs to bind virtual CPUs to,
+potentially impacting NUMA node placement of the guest as a result.
+
+Given a k8s API resource describing the virtual machine, it will be possible to
+generate a libvirt domain XML document from scratch. The particular API usage
+scenario, however, may affect what information is provided in the domain XML.
+The domain XML generated at the time a guest is started, will be the most fully
+specified XML in terms of both frontend and backend configuration. If domain XML
+is generated in isolation, however, while the frontend can be fully specified,
+the backend configuration may remain partially specified. For example, it would
+be possible to expand a PVC to a backend disk config, but without knowledge of
+a specific host to run the guest, the domain XML cannot contain information
+about CPU pinning.
+
+Going in the reverse direction is more difficult in general. The conversion
+from the high level k8s resource to the domain XML format is a lossy process.
+So given a domain XML document in isolation, it is not possible to generate a
+full k8s resource description of the virtual machine. At most it can contain
+frontend information. Translating the backend information in the domain XML, to
+the backend information required in the k8s resource is impractical or impossible
+in many cases. For example, there's no practical way to determine that a disk
+path on a particular host corresponds to a specific PVC. There is none the less
+a need for a conversion from domain XML to the k8s resource, in order to
+preserve the virtual hardware information for subsequent boots. Rather than
+trying to perform this conversion in isolation, however, it is more practical to
+take an initial k8s resource configuration (which contains all the high level
+backend information) and then augment the frontend configuration data with any
+missing pieces.
+
+A further complication is that a single k8s API resource, may end up triggering
+the generation of multiple libvirt XML documents, the domain XML and associated
+documents for secrets, nework filters and node devices. So there is not a strict
+1-1 mapping between the k8s API resource representation and the libvirt low
+level representation.
```

Comment:
If you're talking to an active libvirtd daemon and querying the XML docs, there's no particular problem, as you can ask it for the related xml docs. If you have been given a domain XML doc in isolation though, there's no immediate way to acquire the related docs you might need to interpret it. 

In the case of secret XML objects, however, once you set a value for a secret, libvirt may not allow you to read it back again, as a security measure to prevent future libvirt clients sucking down passwords.